File: F/744-find-smallest-letter-greater-than-target/main.go
package main

import "fmt"

func nextGreatestLetter(letters []byte, target byte) byte {
	s, e := 0, len(letters)-1
	for s < e {
		m := s + (e-s)/2
		if letters[m] <= target {
			s = m + 1
		} else {
			e = m
		}
	}
	if e == len(letters)-1 && letters[e] <= target {
		return letters[0]
	}
	if letters[e] > target {
		return letters[e]
	} else {
		return letters[e+1]
	}
}

func main() {
	fmt.Println("a")
}

File: F/744-find-smallest-letter-greater-than-target/main.rs
impl Solution {
  pub fn next_greatest_letter(letters: Vec<char>, target: char) -> char {
    let mut s: usize = 0;
    let mut e: usize = letters.len();
    while s < e {
      let m = s + (e - s) / 2;
      if letters[m] > target {
        e = e - 1;
      } else {
        s = s + 1;
      }
    }
    if s == letters.len() {
      letters[0]
    } else {
      letters[s]
    }
  }
}
